### PR TITLE
fix(components): el-select-allow-dynamically-update-options

### PR DESCRIPTION
**Pull request by gjfei**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by btea** [comment]
@gjfei The test seems to have failed, please check it.

**Comment by JeremyWuuuuu** [comment]
@gjfei Thank you for contributing, next time let's write the title properly to make it human readable

File: packages/components/select/__tests__/select.spec.ts [hunk 1]
```diff
@@ -729,6 +729,45 @@ describe('Select', () => {
...
+        </el-option>
+      </el-select>
+    `,
+      () => ({
+        options: [],
+        value: '选项2',
+      })
+    )
+    await nextTick()
+    await wrapper.setData({
```
Comment (JeremyWuuuuu) [inline packages/components/select/__tests__/select.spec.ts:764]: Let's assert the length before line#764:

```ts
expect(getOptions()).toHaveLength(0)
```

File: packages/components/select/__tests__/select.spec.ts [hunk 2]
```diff
@@ -729,6 +729,45 @@ describe('Select', () => {
...
+      options,
+    })
+
+    expect(getOptions().length).toBe(options.length)
```
Comment (JeremyWuuuuu) [inline packages/components/select/__tests__/select.spec.ts:768]:
```suggestion
    expect(getOptions()).toHaveLength(options.length)
```

**Comment by JeremyWuuuuu** [comment]
I have left a few comments for you to consider, thank you for contributing to Element Plus, if you have any questions, let me know :pray:

File: packages/components/select/__tests__/select.spec.ts (hunk 1 above)
Comment (gjfei) [reply packages/components/select/__tests__/select.spec.ts:766]: 这里应该是1 不是0

**Review by jw-foss — APPROVED** [review]

**Pull request merged**